src/alerts.rs

```diff
@@ -91,8 +102,22 @@ impl<H: Hasher, D: Data, S: Signature> Signable for Alert<H, D, S> {
     }
 }
 
+impl<H: Hasher, D: Data, S: Signature> PartialEq for Alert<H, D, S> {
+    fn eq(&self, other: &Self) -> bool {
+        self.key() == other.key()
+    }
+}
```

Comment:
Also, there is a `derivative` crate, which allows you to add an attribute `#[derivative(PartialEq="ignore")]` to ignore a field, we can consider using it.

Reply:
Ah, I've been looking for something like `derivative` in this context, added and thanks!

I added the method not because it was more readable (which it wasn't), but to keep the implementations of `Hash` and `PartialEq` in sync. Thankfully this is much simpler with `derivative`, so the method is unnecessary. :>